pkg/build/controller/factory/factory.go

```diff
@@ -14,16 +14,17 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	builder "github.com/openshift/origin/pkg/build/builder"
 	controller "github.com/openshift/origin/pkg/build/controller"
 	strategy "github.com/openshift/origin/pkg/build/controller/strategy"
-	buildutil "github.com/openshift/origin/pkg/build/util"
 	osclient "github.com/openshift/origin/pkg/client"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
 type BuildControllerFactory struct {
-	Client              *osclient.Client
-	KubeClient          *kclient.Client
+	OSClient            osclient.Interface
+	KubeClient          kclient.Interface
+	BuildClient         builder.BuildClientInterface
```

Comment:
The logic behind this is that BuildControllerFactory creates BuildControllers, BuildControllers need to update Builds.  To provide abstraction for testing, we need to be able to create BuildControllerFactories that will create BuildControllers that use arbitrary mechanism to update Builds.  To accomplish that, I've provided a way to pass in an arbitrary BuildClient implementation.  

This allows for testing of the entire Factory implementation without having to mock an entire osclient.

As for locating the canonical implementation with the consumer, I can move the implementation into the factory, but master.go hardly seems like the right place (which is where there canonical implementation was located before)